processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -209,6 +209,20 @@ func New(
 // Start registers pod event handlers and starts watching the kubernetes cluster for pod changes.
 func (c *WatchClient) Start() error {
 	synced := make([]cache.InformerSynced, 0)
+
+	if c.Rules.DeploymentName || c.Rules.DeploymentUID {
```

Comment:
nit: I would add a comment here to note that order is important.

Reply:
good point, I have added a comment now